### PR TITLE
Change module slug/directory from `dominant-color` to `dominant-color-images`

### PR DESCRIPTION
**Pull request by felixarntz**
## Summary

Follow up to #705.
Related to #640.

## Relevant technical choices

* This PR renames the module directory (which is also the module slug) and updates references accordingly.
* It includes the old slug under legacy modules so that the data for sites that have the module active is correctly migrated.
* The PR also changes the module description of Dominant Color Images based on feedback from https://github.com/WordPress/performance/pull/704#pullrequestreview-1376128002.
* Generated files and `readme.txt` are not updated as part of this PR, as those will be updated automatically as part of the release process, except for `default-enabled-modules.php`, which is used by unit tests and needed to be updated so that they still work correctly.

## Checklist

- [x] PR has either `[Focus]` or `Infrastructure` label.
- [x] PR has a `[Type]` label.
- [x] PR has a milestone or the `no milestone` label.

### CODE REVIEW

**Comment by felixarntz** [comment]
cc @mukeshpanchal27 @10upsimon Let's aim to get this merged for the 2.2.0 release as well so that it goes out together with #705. This means we should do some extra testing in this release that, when using an older plugin version and we have the module active it remains active after the update to this release.

I'll be OOO next week, but please feel free to take over this PR, should it require any updates I don't see today.

File: tests/modules/images/dominant-color-images/dominant-color-image-editor-gd-test.php [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?php
 /**
- * Tests for dominant-color module.
+ * Tests for Dominant Color Images module.
  *
  * @package performance-lab
  * @group   dominant-color
```
Comment (mukeshpanchal27) [inline tests/modules/images/dominant-color-images/dominant-color-image-editor-gd-test.php]:
```suggestion
 * @group   dominant-color-images
```

Comment (mukeshpanchal27) [reply tests/modules/images/dominant-color-images/dominant-color-image-editor-gd-test.php]: Addressed in https://github.com/WordPress/performance/pull/708/commits/68fa2df94d143ddbba86815232e32e502299bd9e

**Review by mukeshpanchal27 — APPROVED** [review]
Thanks @felixarntz, The changes look good to me. In https://github.com/WordPress/performance/pull/708/commits/68fa2df94d143ddbba86815232e32e502299bd9e, I have updated two `@group` annotation instances.

**Review by 10upsimon — APPROVED** [review]

**Pull request merged**